workflow/controller/controller.go

```diff
@@ -291,15 +291,38 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, workflowTTLWo
 
 	wfc.wfInformer = util.NewWorkflowInformer(wfc.dynamicInterface, wfc.GetManagedNamespace(), workflowResyncPeriod, wfc.tweakListRequestListOptions, wfc.tweakWatchRequestListOptions, indexers)
 	wfc.wftmplInformer = informer.NewTolerantWorkflowTemplateInformer(wfc.dynamicInterface, workflowTemplateResyncPeriod, wfc.managedNamespace)
-	wfc.wfTaskSetInformer = wfc.newWorkflowTaskSetInformer()
-	wfc.artGCTaskInformer = wfc.newArtGCTaskInformer()
-	wfc.taskResultInformer = wfc.newWorkflowTaskResultInformer()
 
-	wfc.addWorkflowInformerHandlers(ctx)
-	wfc.podInformer = wfc.newPodInformer(ctx)
+	var err error
+	wfc.wfTaskSetInformer, err = wfc.newWorkflowTaskSetInformer()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	wfc.artGCTaskInformer, err = wfc.newArtGCTaskInformer()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	wfc.taskResultInformer, err = wfc.newWorkflowTaskResultInformer()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = wfc.addWorkflowInformerHandlers(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	wfc.podInformer, err = wfc.newPodInformer(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
 	wfc.updateEstimatorFactory()
 
-	wfc.configMapInformer = wfc.newConfigMapInformer()
+	wfc.configMapInformer, err = wfc.newConfigMapInformer()
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
so except for the `add*InformerHandler` functions, all of the `new*Informer` functions will never error out, so we can actually skip the error checking.
they [only error](https://github.com/kubernetes/client-go/blob/46588f2726fa3e25b1704d6418190f424f95a990/tools/cache/shared_informer.go#L580) in the case of the informer having been stopped or started, and they get created within these functions (and started lower down in this function), so that will never be the case.


I'm going to write up a separate PR to fix that as I happened to work on that and notice it during the process of #11855.

Reply:
Removed in #12917